File: backend/game-service/src/main/java/com/popplar/gameservice/dto/MemberInfoDto.java
package com.popplar.gameservice.dto;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@ToString
@NoArgsConstructor
public class MemberInfoDto {

    @JsonProperty
    private Long id;
    @JsonProperty
    private String name;
    @JsonProperty
    private String profileImage;

    @Builder
    public MemberInfoDto(Long id, String name, String profileImage) {
        this.id = id;
        this.name = name;
        this.profileImage = profileImage;
    }
}

File: backend/member-service/src/main/kotlin/com/hotspot/member/dto/MemberInfoDto.kt
package com.hotspot.member.dto

import com.hotspot.member.entity.Member

class MemberInfoDto(
        val id: Long? = null,
        var name: String,
        var profileImage: String
) {
    companion object {
        fun create(member: Member): MemberInfoDto {
            return MemberInfoDto(
                    id = member.id,
                    name = member.name,
                    profileImage = member.profileImage
            )
        }
    }
}